dynamic-layers/raspberrypi/recipes-bsp/u-boot/u-boot_%.bbappend

```diff
@@ -14,7 +14,13 @@ OMNECT_BOOTLOADER_CHECKSUM_FILES += "${LAYERDIR_raspberrypi}/recipes-bsp/u-boot/
 OMNECT_BOOTLOADER_CHECKSUM_FILES += "${LAYERDIR_raspberrypi}/recipes-bsp/u-boot/files/*"
 # we have to update the raspberrypi firmware if basic configuration of the bsp changes
 # per convention such changes should be made in the following file:
-OMNECT_BOOTLOADER_CHECKSUM_FILES += "${LAYERDIR_omnect}/conf/machine/include/rpi_firmware_settings.inc"
+OMNECT_BOOTLOADER_CHECKSUM_FILES:raspberrypi4-64 += "${LAYERDIR_omnect}/conf/machine/include/rpi_firmware_settings.inc"
+# also embedding bootloader version influences u-boot binary, so file below has
+# also to be taken into account for version checksumming
+# NOTE:
+#   next and previous additions need to be tagged with raspberrypi4-64
+#   because any devices derived from it can have different settings here
```

Comment:
the above needed refactoring anyway - and I had a lot of fun with that btw ;-/ - because setting it for raspberrypi4-64 is neither the full truth nor does it work as expected: firstly there's also raspberrypi3 (even though not officially supported), and secondly having a machine specific value defined as above unfortunately restarts definition of the variable, only taking into account other settings done the same way; of course, that makes the checksum mechanism pretty useless ...

apart from the above I rephrased it.

Reply:
i don't understand why you make it machine specific, e.g. you don't do that for phytec devices.
imho the file doesn't get parsed if the layer meta-raspberrypi is not present.
(i don't understand the rephrased note as well.)

Reply:
changed as discussed: don't implement special treatment here but let derived devices in other layers take care of taking out irrelevant files from checksum calculation.